source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -58,9 +78,33 @@ ProxyFilter::ProxyFilter(const std::string& stat_prefix, Stats::Scope& scope,
 
 ProxyFilter::~ProxyFilter() { ASSERT(!delay_timer_); }
 
+void ProxyFilter::setDynamicMetadata(std::string op, absl::optional<std::string> db,
+                                     absl::optional<std::string> collection) {
+  ProtobufWkt::Struct metadata;
+  auto& fields = *metadata.mutable_fields();
+  auto& list = *fields[DynamicMetadataKeysSingleton::get().MessagesField].mutable_list_value();
+  auto& message = *list.add_values()->mutable_struct_value()->mutable_fields();
```

Comment:
How is this list cleared? Won't it keep growing? If we are otherwise just overwriting the metadata, how does this work?

Reply:
I still don't understand how this is correct. AFAICT, this is going to make a new metdata on this call, which will overwrite the previous metadata. How will the list ever be of size > 1. Furthermore, if you do a get followed by a set, how will the list ever be guaranteed to be cleared?

Reply:
In my opinion, this is strictly a developer toggle. Mongo should not clear the list. The consuming filter will clear the list after it consumes the entire metadata. Given that its off by default and as you said, no one will ever use it directly, its safe to let the list grow. The mixer telemetry filter will reset it after sending a batch of telemetry data. This will avoid unnecessary data copies and other issues. If we clear this everytime mongo starts decoding, then any other consuming filter (mixer or not) will have to do a deep copy of the metadata - its not really efficient.

@venilnoronha We need to update the documentation above to state that the 'emit dynamic metadata' is strictly a developer-centric feature that is meant for other filters to consume,...

Reply:
Hmm, having something that grows unbounded such that we assume another filter clears it is pretty shady, which is why I think this whole thing is a total hack. It's self-contained enough that I'm not going to fight it that much, but IMO it's better to have this filter auto-clear. I will let the two of you decide. LMK once you have discussed.